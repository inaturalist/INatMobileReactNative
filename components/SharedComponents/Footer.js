// @flow strict-local

import * as React from "react";
import { Pressable, Text, View } from "react-native";
import { viewStyles } from "../../styles/sharedComponents/footer";

const Footer = ( ): React.Node => (
  <View style={[viewStyles.row, viewStyles.shadow]}>
    <Pressable>
      <Text>menu</Text>
    </Pressable>
    <Pressable>
      <Text>explore</Text>
    </Pressable>
    <Pressable>
      <Text>camera</Text>
    </Pressable>
    <Pressable>
      <Text>obs list</Text>
    </Pressable>
    <Pressable>
      <Text>notifications</Text>
    </Pressable>
  </View>
);

export default Footer;
