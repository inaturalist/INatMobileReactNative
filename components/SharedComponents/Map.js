// @flow

import * as React from "react";
import { View } from "react-native";
import MapView from "react-native-maps";
import { viewStyles } from "../../styles/sharedComponents/map";

type Props = {
  latitude: number,
  longitude: number
}

// this component will need a fallback to another map library
// for people who don't use GMaps (i.e. users in China)
const Map = ( { latitude, longitude }: Props ): React.Node => (
  <View style={viewStyles.mapContainer}>
    <MapView
      style={viewStyles.map}
      initialRegion={{
        latitude,
        longitude,
        latitudeDelta: 0.0922,
        longitudeDelta: 0.0421
      }}
    />
  </View>
);

export default Map;
