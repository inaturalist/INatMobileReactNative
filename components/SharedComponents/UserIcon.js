// @flow

import * as React from "react";
import { Image } from "react-native";

import { imageStyles } from "../../styles/obsDetails";

type Props = {
  uri: string,
  large?: boolean
}

const UserIcon = ( { uri, large }: Props ): React.Node => {
  const imageSource = { uri };

  return (
    <Image source={imageSource} style={[
      imageStyles.userProfileIcon,
      large && imageStyles.largeIcon
    ]} />
  );
};

export default UserIcon;
