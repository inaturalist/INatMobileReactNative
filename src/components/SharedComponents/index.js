export { default as ActivityAnimation } from "./ActivityAnimation/ActivityAnimation";
export { default as ActivityCount } from "./ActivityCount/ActivityCount";
export { default as CommentsCount } from "./ActivityCount/CommentsCount";
export { default as IdentificationsCount } from "./ActivityCount/IdentificationsCount";
export { default as ActivityIndicator } from "./ActivityIndicator";
export { default as BackButton } from "./Buttons/BackButton";
export { default as Button } from "./Buttons/Button";
export { default as CloseButton } from "./Buttons/CloseButton";
export { default as EvidenceButton } from "./Buttons/EvidenceButton";
export { default as INatIconButton } from "./Buttons/INatIconButton";
export { default as RotatingINatIconButton } from "./Buttons/RotatingINatIconButton";
export { default as TransparentCircleButton } from "./Buttons/TransparentCircleButton";
export { default as Checkbox } from "./Checkbox";
export { default as ConfidenceInterval } from "./ConfidenceInterval";
export { default as DateDisplay } from "./DateDisplay";
export { default as DateTimePicker } from "./DateTimePicker";
export { default as DetailsMap } from "./DetailsMap";
export { default as DisplayTaxon } from "./DisplayTaxon";
export { default as DisplayTaxonName } from "./DisplayTaxonName";
export { default as Divider } from "./Divider/Divider";
export { default as FloatingActionBar } from "./FloatingActionBar";
export { default as HideView } from "./HideView";
export { default as IconicTaxonChooser } from "./IconicTaxonChooser";
export { default as IconicTaxonIcon } from "./IconicTaxonIcon";
export { default as INatIcon } from "./INatIcon";
export { default as InlineUser } from "./InlineUser/InlineUser";
export { default as InputField } from "./InputField";
export { default as KebabMenu } from "./KebabMenu";
export { default as LocationPermissionGate } from "./LocationPermissionGate";
export { default as Map } from "./Map";
export { default as MediaNavButtons } from "./MediaNavButtons";
export { default as Modal } from "./Modal";
export { default as Mortal } from "./Mortal";
export { default as ObservationLocation } from "./ObservationLocation";
export { default as ObservationsFlashList } from "./ObservationsFlashList/ObservationsFlashList";
export { default as ObsStatus } from "./ObsStatus";
export { default as OverviewCounts } from "./OverviewCounts";
export { default as PermissionGate } from "./PermissionGate";
export { default as PhotoCount } from "./PhotoCount";
export { default as ProjectListItem } from "./ProjectListItem";
export { default as QualityGradeStatus } from "./QualityGradeStatus/QualityGradeStatus";
export { default as RadioButtonRow } from "./RadioButtonRow";
export { default as ScrollableWithStickyHeader } from "./ScrollableWithStickyHeader";
export { default as ScrollViewWrapper } from "./ScrollViewWrapper";
export { default as SearchBar } from "./SearchBar";
export { default as BottomSheet } from "./Sheets/BottomSheet";
export { default as BottomSheetStandardBackdrop } from "./Sheets/BottomSheetStandardBackdrop";
export { default as PickerSheet } from "./Sheets/PickerSheet";
export { default as RadioButtonSheet } from "./Sheets/RadioButtonSheet";
export { default as TextInputSheet } from "./Sheets/TextInputSheet";
export { default as TextSheet } from "./Sheets/TextSheet";
export { default as WarningSheet } from "./Sheets/WarningSheet";
export { default as StickyToolbar } from "./StickyToolbar";
export { default as Tabs } from "./Tabs/Tabs";
export { default as TaxonResult } from "./TaxonResult";
export { default as Body1 } from "./Typography/Body1";
export { default as Body1Bold } from "./Typography/Body1Bold";
export { default as Body2 } from "./Typography/Body2";
export { default as Body3 } from "./Typography/Body3";
export { default as Body3Bold } from "./Typography/Body3Bold";
export { default as Body4 } from "./Typography/Body4";
export { default as Heading1 } from "./Typography/Heading1";
export { default as Heading2 } from "./Typography/Heading2";
export { default as Heading3 } from "./Typography/Heading3";
export { default as Heading4 } from "./Typography/Heading4";
export { default as Heading5 } from "./Typography/Heading5";
export { default as INatText } from "./Typography/INatText";
export { default as INatTextLight } from "./Typography/INatTextLight";
export { default as INatTextMedium } from "./Typography/INatTextMedium";
export { default as List1 } from "./Typography/List1";
export { default as List2 } from "./Typography/List2";
export { default as P } from "./Typography/P";
export { default as Subheading1 } from "./Typography/Subheading1";
export { default as UploadStatus } from "./UploadStatus/UploadStatus";
export { default as UserIcon } from "./UserIcon/UserIcon";
export { default as UserListItem } from "./UserListItem";
export { default as UserText } from "./UserText";
export { default as ViewWrapper } from "./ViewWrapper";
