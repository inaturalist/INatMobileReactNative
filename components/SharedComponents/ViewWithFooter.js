// @flow strict-local

import * as React from "react";
import { SafeAreaView } from "react-native";

import viewStyles from "../../styles/sharedComponents/viewWithFooter";
import Footer from "./Footer";

type Props = {
  children: React.Node
}

const ViewWithFooter = ( { children }: Props ): React.Node => (
  <SafeAreaView style={viewStyles.safeAreaContainer}>
    {children}
    <Footer />
  </SafeAreaView>
);

export default ViewWithFooter;
