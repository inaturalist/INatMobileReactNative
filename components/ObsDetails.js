// @flow


import React from "react";
import { Text } from "react-native";
import type { Node } from "react";
import ViewWithFooter from "./SharedComponents/ViewWithFooter";
import { ScrollView } from "react-native-gesture-handler";

const ObsDetails = ( ): Node => (
  <ViewWithFooter>
    <ScrollView>
    <Text>username</Text>
    <Text>observation date</Text>
    <Text>observation photos, including faved observations & copywrite info</Text>
    <Text>default photo</Text>
    <Text>species card: rank, common name, sci name, same right hand side as obslist</Text>
    <Text>username</Text>
    <Text>location of obs</Text>
    <Text>two tabs: activity, data</Text>
    <Text>activity: username, computervision, leading/improving, time since comment</Text>
    <Text>default photo with common name/name</Text>
    <Text>comments</Text>
    <Text>agree button</Text>
    <Text>data: description, location map, date observed, date uploaded</Text>
    <Text>data: projects, annotations, obs fields, obs created using, copyright</Text>
    </ScrollView>
  </ViewWithFooter>
);

export default ObsDetails;
