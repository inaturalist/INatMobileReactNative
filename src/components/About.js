// @flow

import { useNavigation } from "@react-navigation/native";
import classnames from "classnames";
import {
  Body1,
  Body2,
  Button,
  Heading4,
  ScrollViewWrapper,
  Tabs,
  ViewWrapper
} from "components/SharedComponents";
import { Image, View } from "components/styledComponents";
import { t } from "i18next";
import type { Node } from "react";
import React, { useState } from "react";
import { getBuildNumber, getVersion } from "react-native-device-info";
import { useDebugMode } from "sharedHooks";

const aboutID = "about";
const teamID = "team";

const About = (): Node => {
  const navigation = useNavigation();
  const [activeTab, setActiveTab] = useState( aboutID );
  const appVersion = getVersion();
  const buildVersion = getBuildNumber();
  const { isDebug, toggleDebug } = useDebugMode( );

  const onTermsPressed = async () => {
    const url = "https://www.inaturalist.org/pages/terms";
    navigation.navigate( "FullPageWebView", {
      title: t( "Terms-of-Use" ),
      initialUrl: url,
      loggedIn: false,
      openLinksInBrowser: true
    } );
  };

  const onPrivacyPressed = async () => {
    const url = "https://www.inaturalist.org/pages/privacy";
    navigation.navigate( "FullPageWebView", {
      title: t( "Privacy-Policy" ),
      initialUrl: url,
      loggedIn: false,
      openLinksInBrowser: true
    } );
  };

  const onCommunityPressed = async () => {
    const url = "https://www.inaturalist.org/pages/community+guidelines";
    navigation.navigate( "FullPageWebView", {
      title: t( "Community-Guidelines" ),
      initialUrl: url,
      loggedIn: false,
      openLinksInBrowser: true
    } );
  };

  return (
    <ViewWrapper>
      <Tabs
        tabs={[
          {
            id: aboutID,
            text: t( "ABOUT" ),
            onPress: () => {
              setActiveTab( aboutID );
            }
          },
          {
            id: teamID,
            text: t( "TEAM" ),
            onPress: () => {
              setActiveTab( teamID );
            }
          }
        ]}
        activeId={activeTab}
      />
      <ScrollViewWrapper>
        {activeTab === aboutID && (
          <View className="px-4 py-8">
            <View className="mb-8 items-center justify-center">
              <Image
                className="w-[250px] h-[45px]"
                resizeMode="contain"
                source={require( "images/inaturalist-dark.png" )}
                accessibilityIgnoresInvertColors
              />
            </View>
            <Heading4 className="mb-3">
              {t( "INATURALIST-MISSION-VISION" )}
            </Heading4>
            <Body2 className="mb-5">
              {t( "iNaturalist-mission-is-to-connect" )}
            </Body2>
            <Body2 className="mb-8">{t( "iNaturalists-vision-is-a-world" )}</Body2>
            <Heading4 className="mb-3">{t( "WHAT-IS-INATURALIST" )}</Heading4>
            <Body2 className="mb-5">
              {t( "iNaturalist-helps-you-identify" )}
            </Body2>
            <Body2 className="mb-5">{t( "Whats-more-by-recording" )}</Body2>
            <Body2 className="mb-8">{t( "iNaturalist-is-supported-by" )}</Body2>
            <Body2
              className="mb-5 underline font-bold"
              onPress={() => onTermsPressed()}
            >
              {t( "Terms-of-Use" )}
            </Body2>
            <Body2
              className="mb-5 underline font-bold"
              onPress={() => onPrivacyPressed()}
            >
              {t( "Privacy-Policy" )}
            </Body2>
            <Body2
              className="mb-8 underline font-bold"
              onPress={() => onCommunityPressed()}
            >
              {t( "Community-Guidelines" )}
            </Body2>
            <View className="items-center justify-center">
              <Body1>{`Version ${appVersion} (${buildVersion})`}</Body1>
            </View>
            <Button
              // eslint-disable-next-line multiline-ternary
              text={`Debug Mode: ${isDebug ? "on" : "off"}`}
              className={classnames(
                "mt-5",
                isDebug
                  ? "bg-deeppink"
                  : "border-deeppink"
              )}
              level={
                isDebug
                  ? "primary"
                  : null
              }
              onPress={toggleDebug}
            />
          </View>
        )}
        {activeTab === teamID && (
          <View className="px-4 py-8">
            <Heading4 className="mb-3">{t( "INATURALIST-TEAM" )}</Heading4>
            <Body2 className="mb-8">
              {t( "iNaturalists-apps-are-designed-and-developed" )}
            </Body2>
            <Heading4 className="mb-3">{t( "INATURALIST-COMMUNITY" )}</Heading4>
            <Body2 className="mb-8">
              {t( "iNaturalist-is-supported-by-community" )}
            </Body2>
            <Heading4 className="mb-3">{t( "INATURALIST-NETWORK" )}</Heading4>
            <Body2 className="mb-8">{t( "The-iNaturalist-Network" )}</Body2>
            <Heading4 className="mb-3">{t( "COLLABORATORS" )}</Heading4>
            <Body2>{t( "The-iNaturalist-team-has-collaborated" )}</Body2>
          </View>
        )}
      </ScrollViewWrapper>
    </ViewWrapper>
  );
};

export default About;
