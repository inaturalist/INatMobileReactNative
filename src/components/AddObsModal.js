// @flow

import { useNavigation } from "@react-navigation/native";
import classnames from "classnames";
import {
  Body3, Heading2, INatIcon, INatIconButton
} from "components/SharedComponents";
import { View } from "components/styledComponents";
import { getCurrentRoute } from "navigation/navigationUtils";
import * as React from "react";
import { Platform, StatusBar } from "react-native";
import { useTheme } from "react-native-paper";
import Observation from "realmModels/Observation";
import { useTranslation } from "sharedHooks";
import useStore from "stores/useStore";

type Props = {
  closeModal: ( ) => void
}

const AddObsModal = ( { closeModal }: Props ): React.Node => {
  const { t } = useTranslation( );
  const theme = useTheme( );
  const setObservations = useStore( state => state.setObservations );
  const resetStore = useStore( state => state.resetStore );

  const majorVersionIOS = parseInt( Platform.Version, 10 );

  // TODO: update these version numbers based on what the new model can handle
  // in CoreML and TFLite
  const showARCamera = ( Platform.OS === "ios" && majorVersionIOS >= 11 )
    || ( Platform.OS === "android" && Platform.Version > 23 );

  const navigation = useNavigation( );

  const navAndCloseModal = async ( screen, params ) => {
    const currentRoute = getCurrentRoute();
    resetStore( );
    if ( screen === "ObsEdit" ) {
      const newObservation = await Observation.new( );
      setObservations( [newObservation] );
    }
    // access nested screen
    navigation.navigate( "CameraNavigator", {
      screen,
      params: { ...params, previousScreen: currentRoute }
    } );
    closeModal( );
  };

  const navToPhotoGallery = async ( ) => {
    navAndCloseModal( "PhotoGallery" );
  };

  const navToSoundRecorder = ( ) => navAndCloseModal( "SoundRecorder" );

  const navToARCamera = ( ) => navAndCloseModal( "Camera", { camera: "AR" } );

  const navToStandardCamera = ( ) => navAndCloseModal( "Camera", { camera: "Standard" } );

  const navToObsEdit = ( ) => navAndCloseModal( "ObsEdit" );

  const bulletedText = [
    { text: t( "Use-iNaturalists-AI-Camera" ), icon: "arcamera" },
    { text: t( "Take-photos-with-the-camera" ), icon: "camera" },
    { text: t( "Upload-photos-from-your-gallery" ), icon: "gallery" },
    { text: t( "Record-sounds" ), icon: "microphone" },
    { text: t( "Create-an-observation-evidence" ), icon: "noevidence" }
  ];

  const greenCircleClass = "bg-inatGreen rounded-full h-[46px] w-[46px]";

  return (
    <>
      <StatusBar barStyle="light-content" backgroundColor="black" />
      <View className="flex-row justify-center">
        <View className="bg-white rounded-xl p-6 mb-12 mx-7 max-w-sm">
          <Heading2 testID="identify-text">
            {t( "Identify-an-organism" )}
          </Heading2>
          {bulletedText.map( ( { text, icon } ) => (
            <View key={text} className="flex-row items-center mt-4">
              <INatIcon
                name={icon}
                size={30}
                color={
                  icon === "arcamera"
                    ? theme.colors.secondary
                    : theme.colors.primary
                }
              />
              <Body3 className="ml-5">{text}</Body3>
            </View>
          ) )}
        </View>
      </View>
      <View className={classnames( "flex-row justify-center", {
        "bottom-[20px]": !showARCamera
      } )}
      >
        <INatIconButton
          testID="camera-button"
          size={30}
          icon="camera"
          color={theme.colors.onSecondary}
          className={classnames( greenCircleClass, {
            "mr-[37px] bottom-[1px]": showARCamera,
            "mr-[9px]": !showARCamera
          } )}
          onPress={navToStandardCamera}
          accessibilityLabel={t( "Camera" )}
          accessibilityHint={t( "Navigates-to-camera" )}
        />
        {showARCamera && (
          <INatIconButton
            testID="arcamera-button"
            size={30}
            icon="arcamera"
            color={theme.colors.onSecondary}
            className={classnames( greenCircleClass, "absolute bottom-[26px]" )}
            onPress={navToARCamera}
            accessibilityLabel={t( "AR-Camera" )}
            accessibilityHint={t( "Navigates-to-AR-camera" )}
          />
        )}
        <INatIconButton
          testID="import-media-button"
          size={30}
          icon="gallery"
          color={theme.colors.onSecondary}
          className={classnames( greenCircleClass, {
            "ml-[37px] bottom-[1px]": showARCamera,
            "ml-[9px]": !showARCamera
          } )}
          onPress={navToPhotoGallery}
          accessibilityLabel={t( "Photo-importer" )}
          accessibilityHint={t( "Navigate-to-photo-importer" )}
        />
      </View>
      <View className="flex-row justify-center items-center">
        <INatIconButton
          testID="observe-without-evidence-button"
          size={30}
          icon="noevidence"
          color={theme.colors.onSecondary}
          className={classnames( greenCircleClass, {
            "mr-[26px]": showARCamera,
            "mr-[20px] bottom-[33px]": !showARCamera
          } )}
          onPress={navToObsEdit}
          accessibilityLabel={t( "Observation-with-no-evidence" )}
          accessibilityHint={t( "Navigate-to-observation-edit-screen" )}
        />
        <INatIconButton
          testID="close-camera-options-button"
          icon="close"
          color={theme.colors.onSecondary}
          size={31}
          className="h-[69px] w-[69px] bg-inatGreen rounded-full"
          onPress={( ) => closeModal( )}
          accessibilityLabel={t( "Close" )}
          accessibilityHint={t( "Close-add-observation-modal" )}
        />
        <INatIconButton
          testID="record-sound-button"
          size={30}
          icon="microphone"
          color={theme.colors.onSecondary}
          className={classnames( greenCircleClass, {
            "ml-[26px]": showARCamera,
            "ml-[20px] bottom-[33px]": !showARCamera
          } )}
          onPress={navToSoundRecorder}
          accessibilityLabel={t( "Sound-recorder" )}
          accessibilityHint={t( "Navigates-to-sound-recorder" )}
        />
      </View>
    </>
  );
};

export default AddObsModal;
