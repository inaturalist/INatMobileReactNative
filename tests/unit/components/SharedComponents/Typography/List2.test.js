import { render, screen } from "@testing-library/react-native";
import { List2 } from "components/SharedComponents";
import React from "react";

const text = "List2";

describe( "List2", () => {
  it( "renders correctly", async () => {
    render( <List2>{text}</List2> );
    // Snapshot test
    expect( screen ).toMatchSnapshot();
  } );
} );
