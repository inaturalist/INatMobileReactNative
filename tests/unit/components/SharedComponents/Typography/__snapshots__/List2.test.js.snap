// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List2 renders correctly 1`] = `
<Text
  style={
    [
      {
        "fontFamily": "Whitney-Book",
      },
      [
        {
          "fontSize": 14,
          "lineHeight": 18,
        },
        {
          "lineHeight": 17,
        },
        {
          "fontWeight": "300",
        },
        {
          "color": "#454545",
        },
      ],
    ]
  }
>
  List2
</Text>
`;
